### PR TITLE
ClassifierResult: add convenience getClass returning most probable label/classification

### LINKED ISSUE #294: Classifier improve max probable label selection
For example, how is the classifier used in MNIST example, 

```
    // Check results
    for(auto iter : result) {
      if( iter.first == 0 ) {
          const auto *pdf = iter.second;
          const auto max  = std::max_element(pdf->cbegin(), pdf->cend());
          const UInt cls  = max - pdf->cbegin();
          if(cls == label)
            score += 1;
          n_samples += 1;
      }
    }
```

add `Classifier::getMaxClass()` which would return `cls` here.

### PR DESCRIPTION
**Pull request by breznak**
- cleanup of Classifier, CLassifierResult
- add UInt ClassifierResult::getClass() 

Fixes #294

### CODE REVIEW

**Comment by breznak** [comment]
Please consider reviewing on per-commit basis, as the first commits change a lot of lines (reorganization), each commit is atomic, so it makes sense on its own. 

Main is the last commit adding getClass()

File: src/nupic/types/ClassifierResult.hpp [hunk 1]
```diff
@@ -33,6 +33,7 @@ namespace nupic {
 namespace types {
 
 using namespace std;
+using T = vector<Real64>;
```
Comment (ctrl-z-9000-times) [inline src/nupic/types/ClassifierResult.hpp]: Question: Is there a significant difference between this and a typedef?   I've never seen this syntax before.  I'm just curious, no need to change it.

Nitpick: Either give it a more descriptive name or add a comment.
Maybe name it `PDF` for probability density function?

File: src/nupic/algorithms/SDRClassifier.hpp [hunk 2]
```diff
@@ -33,16 +33,15 @@
...
 #include <nupic/types/Types.hpp>
 #include <nupic/types/Serializable.hpp>
 
 namespace nupic {
 namespace algorithms {
-
-typedef cla_classifier::ClassifierResult ClassifierResult;
-
 namespace sdr_classifier {
+using namespace std;
```
Comment (ctrl-z-9000-times) [inline src/nupic/algorithms/SDRClassifier.hpp]: Nitpick: `using namespace std` in header?  Is this really needed here?  Usually this line is only in the source files...

File: bindings/py/cpp_src/bindings/algorithms/py_SDRClassifier.cpp [hunk 3]
```diff
@@ -35,12 +35,14 @@ PyBind11 bindings for SDRClassifier class
 
 #include <nupic/algorithms/SDRClassifier.hpp>
 
+
+namespace nupic_ext
+{
 namespace py = pybind11;
 using namespace nupic;
 using namespace nupic::algorithms::sdr_classifier;
+using T = std::vector<Real64>;
```
Comment (ctrl-z-9000-times) [inline bindings/py/cpp_src/bindings/algorithms/py_SDRClassifier.cpp]: Nitpick: can this import the symbol `T` from the header file?

File: src/nupic/types/ClassifierResult.hpp (hunk 1 above)
Comment (breznak) [reply src/nupic/types/ClassifierResult.hpp]: Here it's the same, `using T = ` is a newer, c++11+ construct. 
https://stackoverflow.com/questions/10747810/what-is-the-difference-between-typedef-and-using-in-c11 

> Maybe name it PDF for probability density function? 

good point, I didn't like the T either

File: src/nupic/algorithms/SDRClassifier.hpp (hunk 2 above)
Comment (breznak) [reply src/nupic/algorithms/SDRClassifier.hpp]: true. I didn't add that one, just moved it from global scope to within the ::nupic::algorithms::sdr_classifier

File: bindings/py/cpp_src/bindings/algorithms/py_SDRClassifier.cpp (hunk 3 above)
Comment (breznak) [reply bindings/py/cpp_src/bindings/algorithms/py_SDRClassifier.cpp]: I had hoped so, but it didn;t. I'll investigate if it works.

**Comment by breznak** [comment]
Thanks for review, all feedback should be incorporated in the latest commit.

**Comment by ctrl-z-9000-times** [comment]
Wait one last thing: Does the change to change from pointer to reference for passing ClassifierResults break API compat? If so it should be documented.

**Comment by breznak** [comment]
> Wait one last thing: Does the change to change from pointer to reference for passing ClassifierResults break API compat? If so it should be documented. 

Good spot! I had to change the py bindings, but from the side of python all remains the same. To err on the safe side, I'll add the note to API_breakage

**Comment by breznak** [comment]
added the doc.

**Pull request merged**